src/inmanta/util.py

```diff
@@ -200,7 +201,20 @@ def get_free_tcp_port() -> str:
         return str(port)
 
 
-def custom_json_encoder(o: object) -> Union[Dict, str, List]:
+class JSONSerializable(ABC):
```

Comment:
Please document when to use this class and how to use it. Also document its methods